agent/proxycfg/state.go

```diff
@@ -583,6 +583,8 @@ func (s *state) initialConfigSnapshot() ConfigSnapshot {
 }
 
 func (s *state) run() {
+	logger := s.logger.Named(logging.ProxyConfig)
```

Comment:
non blocking idea:

Use the `With()` call here and tack on the proxy-id logging kv here so that all messages emitted by the logger automatically get proxy-id attached

Reply:
Ah, didn't know about `.With()`, done

Reply:
Actually the labeling was already taken care of: 
https://github.com/hashicorp/consul/blob/a4690ac7d9af43e596d34e9b8c51254a5bfb8e1f/agent/agent.go#L543
https://github.com/hashicorp/consul/blob/a4690ac7d9af43e596d34e9b8c51254a5bfb8e1f/agent/proxycfg/manager.go#L196


Removed it from here.